### PR TITLE
[mlir][LLVM] erase call mappings in forgetMapping()

### PR DESCRIPTION
**Pull request by tblah**
It looks like the mappings for call instructions were forgotten here. This fixes a bug in OpenMP when in-lining a region containing call operations multiple times.

OpenMP array reductions 4/6
Previous PR: https://github.com/llvm/llvm-project/pull/84954
Next PR: https://github.com/llvm/llvm-project/pull/84957

### CODE REVIEW

**Review by kiranchandramohan — APPROVED** [review]
LGTM. Please wait a day incase others have comments.

**Review by Dinistro — APPROVED** [review]

**Comment by joker-eph** [comment]
We need a test here

File: mlir/test/Target/LLVMIR/openmp-reduction-call.mlir [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      %3 = llvm.fadd %1, %2 : f32
+      llvm.store %3, %prv : f32, !llvm.ptr
+      omp.yield
+    }
+    omp.terminator
+  }
+  llvm.return
+}
+
+llvm.func @test_call() -> ()
```
Comment (joker-eph) [inline mlir/test/Target/LLVMIR/openmp-reduction-call.mlir:42]: This seems quite complex to me: is this the minimum test? 

Why do we need any OpenMp constructs here?

File: mlir/test/Target/LLVMIR/openmp-reduction-call.mlir [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  llvm.call @test_call() : () -> ()
+  %1 = llvm.fadd %arg0, %arg1 : f32
+  omp.yield (%1 : f32)
+}
+atomic {
+^bb2(%arg2: !llvm.ptr, %arg3: !llvm.ptr):
+  %2 = llvm.load %arg3 : !llvm.ptr -> f32
+  llvm.atomicrmw fadd %arg2, %2 monotonic : !llvm.ptr, f32
+  omp.yield
+}
```
Comment (kiranchandramohan) [inline mlir/test/Target/LLVMIR/openmp-reduction-call.mlir:23]: Can the atomic region be removed?

File: mlir/test/Target/LLVMIR/openmp-reduction-call.mlir [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+llvm.func @simple_reduction(%lb : i64, %ub : i64, %step : i64) {
+  %c1 = llvm.mlir.constant(1 : i32) : i32
+  %0 = llvm.alloca %c1 x i32 : (i32) -> !llvm.ptr
+  omp.parallel {
+    omp.wsloop reduction(@add_f32 %0 -> %prv : !llvm.ptr)
```
Comment (kiranchandramohan) [inline mlir/test/Target/LLVMIR/openmp-reduction-call.mlir:29]: Can the reduction be moved to `omp.parallel`? And the `omp.wsloop` removed?

File: mlir/test/Target/LLVMIR/openmp-reduction-call.mlir (hunk 1 above)
Comment (kiranchandramohan) [reply mlir/test/Target/LLVMIR/openmp-reduction-call.mlir:42]: > Why do we need any OpenMp constructs here?

`forgetMapping` is only used in the OpenMP translation.

**Review by kiranchandramohan — COMMENTED** [review]
Some minor comments about the test.

**Comment by tblah** [comment]
Sorry about the force push. Github wasn't showing the new commit (02550e1). It is just a rebase onto the target branch.

File: mlir/test/Target/LLVMIR/openmp-reduction-call.mlir (hunk 1 above)
Comment (joker-eph) [reply mlir/test/Target/LLVMIR/openmp-reduction-call.mlir:32]: The OpenMP translation predates the LLVMTranslationInterface, but it really should evolve to this model: the OpenMP specific component should be better identified and isolated really.
Anyway that's not a thing for this PR, thank for clarifying!

**Pull request merged**